core/consensus/grandpa/vote_graph/vote_graph_impl.cpp

```diff
@@ -331,12 +340,15 @@ namespace kagome::consensus::grandpa {
           descendent_blocks[d_block] += entry.cumulative_vote;
           // check if block fullfills condition
           if (condition(descendent_blocks[d_block])) {
+            if (not new_best_vote_weight
+                or new_best_vote_weight < descendent_blocks[d_block]) {
```

Comment:
Didn't know we can compare `boost::optional<VoteWeight>` and `VoteWeight`

Reply:
Yes, we can. From bost sources:
```
template<class T>
inline bool operator < ( optional<T> const& x, T const& y )
{ return less_pointees(x, optional<T>(y)); }
```